File: bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/base/ZeroMagnitudeStar.java
package edu.gemini.itc.base;

import edu.gemini.spModel.core.MagnitudeBand;

/**
 * This class encapsulates the photon flux density (in photons/s/m^2/nm)
 * for a zero-magnitude star.
 */
public final class ZeroMagnitudeStar {
    // Keep anyone from instantiating this object.
    private ZeroMagnitudeStar() {}

    /**
     * Get average flux in photons/s/m^2/nm in specified waveband.
     * Overrides method in base class AverageFlux.
     */
    public static int getAverageFlux(final MagnitudeBand band) {

        // flux densities for each waveband for zero-magnitude star (of some type)
        // units are photons/s/m^2/nm
        // TODO: Missing support for Y band
        if      (band.equals(MagnitudeBand.U$.MODULE$))  return 75900000;
        else if (band.equals(MagnitudeBand.B$.MODULE$))  return 146100000;
        else if (band.equals(MagnitudeBand.V$.MODULE$))  return 97100000;
        else if (band.equals(MagnitudeBand.R$.MODULE$))  return 64600000;
        else if (band.equals(MagnitudeBand.I$.MODULE$))  return 39000000;
        else if (band.equals(MagnitudeBand.J$.MODULE$))  return 19700000;
        else if (band.equals(MagnitudeBand.H$.MODULE$))  return 9600000;
        else if (band.equals(MagnitudeBand.K$.MODULE$))  return 4500000;
        else if (band.equals(MagnitudeBand.L$.MODULE$))  return 990000;
        else if (band.equals(MagnitudeBand.M$.MODULE$))  return 510000;
        else if (band.equals(MagnitudeBand.N$.MODULE$))  return 51000;
        else if (band.equals(MagnitudeBand.Q$.MODULE$))  return 7700;

        // Values for Sloan filters taken from Schneider, Gunn, & Hoessel (1983)
        // TODO: Missing support for u' band
        else if (band.equals(MagnitudeBand._g$.MODULE$)) return 117000000;
        else if (band.equals(MagnitudeBand._r$.MODULE$)) return 108000000;
        else if (band.equals(MagnitudeBand._i$.MODULE$)) return 93600000;
        else if (band.equals(MagnitudeBand._z$.MODULE$)) return 79800000;

        else throw new IllegalArgumentException("ITC does not support waveband " + band.name());

    }
}

File: bundle/edu.gemini.itc/src/main/scala/edu/gemini/itc/base/ZeroMagnitudeStar.scala
package edu.gemini.itc.base

import edu.gemini.spModel.core.MagnitudeBand

/**
 * This class encapsulates the photon flux density (in photons/s/m^2/nm)
 * for a zero-magnitude star.
 */
object ZeroMagnitudeStar {

  import MagnitudeBand._

  /**
   * Get average flux in photons/s/m^2/nm in specified waveband.
   * Overrides method in base class AverageFlux.
   */
  def getAverageFlux(band: MagnitudeBand): Int = band match {

    // flux densities for each waveband for zero-magnitude star (of some type)
    // units are photons/s/m^2/nm
    case U => 75900000
    case B => 146100000
    case V => 97100000
    case R => 64600000
    case I => 39000000
    case Y => 29680000
    case J => 19700000
    case H => 9600000
    case K => 4500000
    case L => 990000
    case M => 510000
    case N => 51000
    case Q => 7700

    // Values for Sloan filters taken from Schneider, Gunn, & Hoessel (1983)
    case `_u` => 65670000
    case `_g` => 117000000
    case `_r` => 108000000
    case `_i` => 93600000
    case `_z` => 79800000

    // Other bands are not supported, in particular AP and UC
    case _ => sys.error("unsupported waveband " + band.name)


  }
}
